src/ext/preload.js

```diff
@@ -48,12 +48,20 @@ htmx.defineExtension("preload", {
 				// in the future
 				var hxGet = node.getAttribute("hx-get") || node.getAttribute("data-hx-get")
 				if (hxGet) {
+					var previousIndicator = node.getAttribute("hx-indicator");
+					node.setAttribute("hx-indicator", "")
```

Comment:
OK, have thought about this more and I think you are doing the right thing here.

The only thing I would change is that here, rather than setting the value to "", set it to "unset" so that if there is a parent hx-indicator, that is also ignored.

Thinking about it, the ajax request will fire off and be handled asyn, but the code will keep on executing so the resetting of the attribute in the next set of if statements will be run before any other htmx requests get fired off.